pkg/sql/schemachanger/scdeps/exec_deps.go

```diff
@@ -117,14 +118,42 @@ func (d *txnDeps) UpdateSchemaChangeJob(
 	return d.jobRegistry.UpdateJobWithTxn(ctx, id, d.txn, useReadLock, func(
 		txn *kv.Txn, md jobs.JobMetadata, ju *jobs.JobUpdater,
 	) error {
+		var pl *jobspb.Payload
+		maybeSetPayload := func() {
+			if pl == nil {
+				clone := *md.Payload
+				pl = &clone
+			}
+		}
 		setNonCancelable := func() {
-			payload := *md.Payload
-			if !payload.Noncancelable {
-				payload.Noncancelable = true
-				ju.UpdatePayload(&payload)
+			maybeSetPayload()
+			if !pl.Noncancelable {
+				pl.Noncancelable = true
+				ju.UpdatePayload(pl)
+			}
+		}
+		removeDescriptorIDs := func(descIDsToRemove []catid.DescID) error {
+			if len(descIDsToRemove) == 0 {
+				return nil
+			}
+			maybeSetPayload()
+			toRemove := catalog.MakeDescriptorIDSet(descIDsToRemove...)
+			var filtered []catid.DescID
+			for _, id := range pl.DescriptorIDs {
+				if toRemove.Contains(id) {
+					toRemove.Remove(id)
+				} else {
+					filtered = append(filtered, id)
+				}
+			}
+			if !toRemove.Empty() {
+				return errors.AssertionFailedf("schema change job not found")
```

Comment:
it feels like we assume here that `toRemove` is a subset of `pl.DescriptorIDs`, and we error out if this it's not.

I think we can probably do something like 
```
currDescriptorIDs := catalog.MakeDescriptorIDSet(pl.DescriptorIDs)
toRemove := catalog.MakeDescriptorIDSet(descIDsToRemove...)
if !toRemove.Minus(currDescriptorIDs).Empty() {
    error out
}
pl.DescriptorIDs = currDescriptorIDs.Minus(toRemove).Ordered()
```

same applied below for `*TestState`

Reply:
Yeah, the complexity was about not mucking with the ordering in the slice. I'll share the code and do something like this.